### ISSUE TITLE
linter: typescript/dot-notation does not flag bracket-notation access for valid-identifier string keys

### ISSUE THREAD

**Issue by kbrooks** [oxc-project/oxc#22532]
### What version of Oxlint are you using?

1.65.0

### What command did you run?

`./node_modules/.bin/oxlint -c .oxlintrc.json index.ts `  (Also reproduced with `oxlint --deny typescript/dot-notation index.ts` — no config file needed; the rule never fires regardless of how it's enabled.)

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc

  {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "rules": {
      "typescript/dot-notation": "error"
    }
  }
```

### What happened?

Repro file (index.ts):
```ts
  export const obj = { foo: 1, bar: 2 };
  export const a = obj['foo'];
  export const b = obj['bar'];

  interface Shape {
    baz: number;
  }
  export const shape: Shape = { baz: 3 };
  export const c = shape['baz'];

  class C {
    hello = 'world';
  }
  export const inst = new C();
  export const d = inst['hello'];
```

**Expected**: four typescript/dot-notation errors — every bracket-notation access uses a valid JS identifier as the key, all four should be reported the way ESLint's dot-notation and `@typescript-eslint/dot-notation` report them.

**Actual**: Found 0 warnings and 0 errors. Same result with `--deny typescript/dot-notation.`

**Comment by camc314** [comment]
does it work if you supply `--type-aware` ?

**Comment by camc314** [comment]
Closing as not planned. 

You `typescipt/dot-notation` is a type aware rule that require `--type-aware` being enabled for it to work correctly.

I validated locally that running `./node_modules/.bin/oxlint --deny typescript/dot-notation index.ts --type-aware` correctly reports the errors as expected.


Please reach out if this does not fix the issue or if there are any other problems.

Thanks!
